src/Reflection/Argument/Modality.agda

```diff
@@ -0,0 +1,57 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Modalities used in the reflection machinery
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Reflection.Argument.Modality where
+
+open import Data.Product
+import Relation.Nullary.Decidable as Dec
+open import Relation.Nullary.Product using (_×-dec_)
+open import Relation.Binary
+open import Relation.Binary.PropositionalEquality
+
+open import Reflection.Argument.Relevance as Relevance using (Relevance)
+open import Reflection.Argument.Quantity as Quantity using (Quantity)
+
+------------------------------------------------------------------------
+-- Re-exporting the builtins publically
+
+open import Agda.Builtin.Reflection public using (Modality)
+open Modality public
+
+------------------------------------------------------------------------
+-- Operations
+
+relevance : Modality → Relevance
+relevance (modality r _) = r
+
+quantity : Modality → Quantity
+quantity (modality _ q) = q
+
+------------------------------------------------------------------------
+-- Decidable equality
+
+private
+  variable
+    r r′ : Relevance
+    q q′ : Quantity
```

Comment:
Variable declarations typically go at the top of the file underneath (non-public) imports.